tests/test_session.py

```diff
@@ -117,3 +117,6 @@ class TestSession(tb.QueryTestCase):
 
             [['entity', 'user']]
         ])
+
+    # TODO: see if nested WITH block with different modules can be
```

Comment:
Is this TODO still relevant?

Reply:
Absolutely, I'll add the tests that I have shown to you in a separate small PR. Maybe I'll manage to come up with more obscure ways of nesting default modules, I'll need to look at the scopes tests for additional inspiration.